internal/controller/pipelinerun_helper_test.go

```diff
@@ -0,0 +1,496 @@
+							},
+						},
+					},
+				},
+			},
+		}
+		Context("When non pipelinerun object created", func() {
+			It("IsPushPipelineRun should return false", func() {
+				got := IsPushPipelineRun(notPipelineRun)
+				Expect(got).To(BeFalse())
+			})
+			It("IsAnnotationExistInPipelineRun should return false", func() {
+				got := IsAnnotationExistInPipelineRun(notPipelineRun, NotificationPipelineRunAnnotation, NotificationPipelineRunAnnotationValue)
+				Expect(got).To(BeFalse())
+			})
+			It("IsFinalizerExistInPipelineRun should return false", func() {
+				got := IsFinalizerExistInPipelineRun(notPipelineRun, NotificationPipelineRunFinalizer)
+				Expect(got).To(BeFalse())
+			})
+			It("IsPipelineRunEnded should return false", func() {
+				got := IsPipelineRunEnded(notPipelineRun)
+				Expect(got).To(BeFalse())
+			})
+			It("IsPipelineRunEndedSuccessfully should return false", func() {
+				got := IsPipelineRunEndedSuccessfully(notPipelineRun)
+				Expect(got).To(BeFalse())
+			})
+		})
+	})
+	Describe("When results fails to marshal", func() {
+		testErrMmarshelPipelineRun = &tektonv1.PipelineRun{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "pull_request",
+				Namespace: notPipelineRun.Namespace,
+				Labels: map[string]string{
+					PipelineRunTypeLabel: PushPipelineRunTypeValue,
+				},
+				Finalizers: []string{
+					NotificationPipelineRunFinalizer,
+				},
+				Annotations: map[string]string{
+					NotificationPipelineRunAnnotation: NotificationPipelineRunAnnotationValue,
+				},
+			},
+			Spec: tektonv1.PipelineRunSpec{
+				PipelineRef: &tektonv1.PipelineRef{},
+			},
+			Status: tektonv1.PipelineRunStatus{
+				PipelineRunStatusFields: tektonv1.PipelineRunStatusFields{
+					StartTime:      &metav1.Time{Time: time.Now()},
+					CompletionTime: &metav1.Time{Time: time.Now().Add(5 * time.Minute)},
+					Results: []tektonv1.PipelineRunResult{
+						{},
+						{
+							Name:  "wrong key and value",
+							Value: tektonv1.ParamValue{Type: "test"},
+						},
+					},
+				},
+				Status: v1.Status{
+					Conditions: v1.Conditions{
+						apis.Condition{
+							Message: "Tasks Completed: 12 (Failed: 0, Cancelled 0), Skipped: 2",
+							Reason:  "Completed",
+							Status:  "True",
+							Type:    apis.ConditionSucceeded,
+						},
+					},
+				},
+			},
+		}
+		Context("when a push pipelinerun is created, and results cannot be marshaled", func() {
+			It("GetResultsFromPipelineRun should return an error", func() {
+				got, err := GetResultsFromPipelineRun(testErrMmarshelPipelineRun)
+				Expect(got).To(BeNil())
+				Expect(err).To(HaveOccurred())
+				Expect(err.Error()).To(ContainSubstring("failed to marshel results from pipelinerun %s", testErrMmarshelPipelineRun.Name))
+			})
+		})
+	})
+
+	Describe("Successfully adding/removing resources when not exist", func() {
+		BeforeEach(func() {
+			// Create a pull_request pipelinerun
+			prPipelineRun = &tektonv1.PipelineRun{
```

Comment:
Does it have a type at all? The name might need to change.

Reply:
In line 24:
`prPipelineRun *tektonv1.PipelineRun`

Reply:
But here we assign to that variable and I don't think there's anything that actually makes it a pr pipelinerun other than the name.

Reply:
You are right, I have removed the pull_request label, let me fix it